### PR TITLE
Ports: Allow Ports to build once again

### PR DESCRIPTION
**Pull request by archanox**
Rearranged the order of parameters for cmake within the harfbuzz package script to enable building with cmake 3.18.4
Point the config_sub_path to the correct location for each given failing port.
Update the download link for gsl to the "non-mirror" url which appears to be dead and unreliable.

### CODE REVIEW

**Comment by archanox** [comment]
> https://cmake.org/cmake/help/v3.23/manual/cmake.1.html suggests that the current syntax is correct.
> 
When left as is with cmake 3.18.4 (the current version shipped with Ubuntu 20.10) we get the following message when running package.sh

```
Configuring harfbuzz...
+ mkdir -p build
+ sh -c cd build && cmake -DCMAKE_TOOLCHAIN_FILE=/home/pierce/serenity/Build/i686/CMakeToolchain.txt -DHB_HAVE_FREETYPE=ON -DHB_HAVE_ICU=ON ..
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.


CMake Error: The source directory "/home/pierce/serenity/Ports/harfbuzz/harfbuzz-2.8.1/build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
```
After rearranging the params as per this PR we get a successful build
```
Configuring harfbuzz...
+ mkdir -p build
+ sh -c cd build && cmake .. -DCMAKE_TOOLCHAIN_FILE=/home/pierce/serenity/Build/i686/CMakeToolchain.txt -DHB_HAVE_FREETYPE=ON -DHB_HAVE_ICU=ON
-- The C compiler identification is GNU 11.2.0
```
etc etc

File: Ports/diffutils/package.sh [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/usr/bin/env -S bash ../.port_include.sh
 port=diffutils
 version=3.7
+config_sub_path=build-aux/config.sub
```
Comment (linusg) [inline Ports/diffutils/package.sh:4]: Please insert these after `use_fresh_config_sub` consistently, and not sometimes after `version` and sometimes not.

**Review by linusg — CHANGES_REQUESTED** [review]
The cmake change is still not explained in the commit message. Additionally, there are other ports using the same syntax.

**Comment by archanox** [comment]
> The cmake change is still not explained in the commit message. 

Fixing that now.
> Additionally, there are other ports using the same syntax.

Do you have a list of the specific ports in question? I'd like to investigate those further. I do see a mish-mash of build locations specified as parameter and as not, there's varying configurations of where built source is located. All in all, it's a bit of a mess that can be cleaned up iteratively.

I'd like to ship this as a preliminary step in getting Ports working out of the box.

**Pull request merged**